syntax = "proto3";
package suggestions;

service Suggestions {
  rpc CacheOrder (OrderRequest) returns (OrderResponse) {}
  rpc GetSuggestions (OrderRequest) returns (OrderResponse) {}
  rpc ClearOrder (ClearRequest) returns (ClearResponse) {}  // 新加的：清理数据
}

message OrderRequest {
  string user_name = 1;
  repeated Item items = 2;
  repeated int32 vector_clock = 3;
  string order_id = 4;

}

message Item {
  string name = 1;
  int32 quantity = 2;
}

message Book {
  string book_id = 1;
  string title = 2;
  string author = 3;
}

message OrderResponse {
  bool success = 1;
  string message = 2;
  repeated Book suggested_books = 3;
  repeated int32 vector_clock = 4;
}

message ClearRequest {
    string order_id = 1;
    repeated int32 final_vector_clock = 2;  // VCf
}

message ClearResponse {
    bool success = 1;
    string message = 2;
}